### PR TITLE
Add authentication

### PR DESCRIPTION
**Pull request by Maianki**
### Files to be checked 
 - [src/App.js](https://github.com/Maianki/shuttle-shopy-react/compare/add-authentication?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)
 - [src/components/Form/Input/Input.jsx](https://github.com/Maianki/shuttle-shopy-react/compare/add-authentication?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)
 - [src/components/Navbar/Navbar.jsx](https://github.com/Maianki/shuttle-shopy-react/compare/add-authentication?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)
 - [src/components/RequireAuthentication/AuthenticatedRoute.jsx](https://github.com/Maianki/shuttle-shopy-react/compare/add-authentication?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)
 - [src/components/RequireAuthentication/NotAuthenticatedRoute.jsx](https://github.com/Maianki/shuttle-shopy-react/compare/add-authentication?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)
 - [](https://github.com/Maianki/shuttle-shopy-react/compare/add-authentication?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)
 - [src/context/auth-context.js](https://github.com/Maianki/shuttle-shopy-react/compare/add-authentication?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)
 - [src/index.js](https://github.com/Maianki/shuttle-shopy-react/compare/add-authentication?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)
 - [src/pages/Login/Login.jsx](https://github.com/Maianki/shuttle-shopy-react/compare/add-authentication?expand=1#diff-3d74dddefb6e35fbffe3c76ec0712d5c416352d9449e2fcc8210a9dee57dff67)
 - [src/pages/Profile/Profile.jsx](https://github.com/Maianki/shuttle-shopy-react/compare/add-authentication?expand=1#diff-93f068f0ecc02ca613ac870d7f640244f3a4621e12401d31d34d6a28eefa608c)
 - [src/pages/SignupPage/SignUpPage.jsx](https://github.com/Maianki/shuttle-shopy-react/compare/add-authentication?expand=1#diff-49d14eb844b4e3736154279f406e9762af43c9ae40dd9311725a77753b47d53a)
 - [src/reducers/authReducer.js](https://github.com/Maianki/shuttle-shopy-react/compare/add-authentication?expand=1#diff-978cb6ff885969efafb050f3963f42db8388343b0f752803094897fa970c7023)
 
### Things that are not included right now
- persistent login on refresh

### Live Preview
[Click here for live preview](https://shuttle-shopy-react-git-add-authentication-maianki.vercel.app/)
Guest login available

#### Thanks for reviewing

### CODE REVIEW

**Review by vrinda-mahajan — APPROVED** [review]
Gone through the whole code . Everything looks perfect. Your code is highly readable and clean.
The folder structure is also up to the mark. Keep up with the good work👍.

**Comment by Maianki** [comment]
> Gone through the whole code . Everything looks perfect. Your code is highly readable and clean. The folder structure is also up to the mark. Keep up with the good work👍.

Thanks for reviewing @vrinda-mahajan

File: src/components/Navbar/Navbar.jsx [hunk 1]
```diff
@@ -1,57 +1,86 @@
...
-            <li className="list-inline">
-                <Link className="flex-column menu-link" to="/cart">
-                    {/* -- cart -- */}
-                    <img className="navbar-link" src={cartIcon} alt="add to cart icon"/>
-                </Link>
-            </li>    
-        </ul>
+        <li className='list-inline'>
+          <Link className='flex-column menu-link' to='/cart'>
+            {/* -- cart -- */}
```
Comment (foolhardy21) [inline src/components/Navbar/Navbar.jsx:56]: appreciation - use of comments.

File: src/context/auth-context.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import axios from "axios";
+import { createContext, useContext, useReducer } from "react";
+import { authReducer, authInitialState } from "../reducers";
+import { LOGIN_API, SIGNUP_API } from "../constants/apiEndPoints";
```
Comment (foolhardy21) [inline src/context/auth-context.js:4]: appreciation - use of constants.

**Review by foolhardy21 — APPROVED** [review]
good use of react router concepts and code is readable. good to go ahead.

**Comment by Maianki** [comment]
> good use of react router concepts and code is readable. good to go ahead.

Thanks for reviewing @foolhardy21

**Pull request merged**